applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -78,6 +78,47 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
         mPipingIterations = rParameters["max_piping_iterations"].GetInt();
     }
 
+    /**
+     * @brief std vector containing non-converged solutions.
+     * @details Each entry in the std vector is a Kratos Vector containing the solution at a specific non-converged iteration.
+     */
+    std::vector<Vector> mNonconvergedSolutions;
+
+
+    /**
+     * @brief This matrix stores the non-converged solutions
+     * @details The matrix is structured such that each column represents the solution vector at a specific non-converged iteration.
+     */
+    Matrix mNonconvergedSolutionsMatrix;
+
+    /**
+     * @brief Flag indicating whether to store non-converged solutions
+     * @details Only when set to true (by calling the SetUpNonconvergedSolutionsGathering method) will the non-converged solutions at each iteration be stored.
+     */
+    bool mStoreNonconvergedSolutionsFlag = false;
+
+
+    /**
+     * @brief Gets the matrix of non-converged solutions and the DOF set
+     * @return A tuple containing the matrix of non-converged solutions and the DOF set
+     * @details This method returns a tuple where the first element is a matrix of non-converged solutions and the second element is the corresponding DOF set.
+     */
+    std::tuple<Matrix, DofsArrayType> GetNonconvergedSolutions(){
```

Comment:
Do you think it is feasible to add a test for this new behavior? If it is not too much work, that would certainly be worthwhile in my opinion (such that we are sure that this functionality keeps working as it does when the `mStoreNonconvergedSolutionsFlag` is set to true

Reply:
I have added a python test for the nonconverged solution gathering. I think this way is easier than implementing a cpp test as in the `test_GeoMechanisNewtonRaphsonErosionProcess.cpp`.  I this a cpp test would involve touching  `dgeoflow.cpp` to access the solving strategy and activate the flag.

Reply:
The test however has not been included in the small or nightly test suite yet